AdaptiveIndexDefrag/usp_AdaptiveIndexDefrag.sql

```diff
@@ -1,8 +1,8 @@
 -- If you are using AdaptiveIndexDefrag together with the maintenance plans in http://blogs.msdn.com/b/blogdoezequiel/archive/2012/09/18/about-maintenance-plans-grooming-sql-server.aspx
 -- please note that the job that runs AdaptiveIndexDefrag is expecting msdb. As such, change the database context accordingly.
 
--- For deployment in Azure SQL Database, remove or comment the USE statement below.
-USE msdb
+-- For deployment in Azure SQL Database, remove or comment the USE statement below. And please, please don't create user objects in system databases like [msdb].
```

Comment:
Myeah.... I sort of agree with you when it comes to master, but msdb is a little more flexible. It's designed to store maintenance functions and metadata, and it actually needs to be user maintained or it can get out of control (something a lot of folks don't pay attention to). Also, if we were to change this here, we would have to add code to create this database elsewhere and then change all of the other scripts to use it. I'd prefer to keep this in msdb. All the other changes look good though, so if you can pull this one, I will merge the request.

Reply:
Thanks for taking a look @bluefooted :) When I get back from holiday next week I will implement your proposal.

Reply:
@bluefooted Leave AdaptiveIndexDefrag deployed to [msdb] database. Also, I noticed that the DateDiff_Big makes that the script will be SQL2016+